File: Builder/Ruby/PDFReportBuilder.rb
require_relative "Report"
require_relative "IReportBuilder"

class PDFReportBuilder < IReportBuilder
    def buildContent
        if !@report
            @report = Report.new
        end

        @report.content = "Summary"
    end

    def buildFile
        if !@report
            @report = Report.new
        end

        @report.filename = "Report.pdf"
    end

    def getResult
        @report
    end
end

File: Builder/Javascript/PDFReportBuilder.js
import IReportBuilder from "./IReportBuilder.js";
import Report from "./Report.js";

class PDFReportBuilder extends IReportBuilder {
  buildContent() {
    if (!this.report) {
      this.report = new Report();
    }

    this.report.content = "Summary";
  }

  buildFile() {
    if (!this.report) {
      this.report = new Report();
    }

    this.report.filename = "Report.pdf";
  }

  getResult() {
    return this.report;
  }
}

export default PDFReportBuilder;
